src/parameter/parameterobject.cpp

```diff
@@ -4,8 +4,30 @@
 #include "modcontext.h"
 #include "annotation.h"
 
-ParameterObject::ParameterObject() : focus(false)
+ParameterObject::ParameterObject(Context *ctx, const Assignment *assignment, const ValuePtr defaultValue) : focus(false)
```

Comment:
Since `assignment` isn't allowed to be NULL, I'd prefer a `const &` here; makes it a bit more clear how it's being used.

Reply:
sounds good - I by the way have now 2nd and 3rd Edition of Bjarne Stroustrups "The C++ Programming Language"-  thank you for the recommendation. I know that there is a 4th edition, but the basics like this are still the same.